### PR TITLE
feat(DataTable): basic DataTable

### PR DESCRIPTION
**Pull request by marekdedic**
First step towards #58 

I'd like to get the basic DataTable merged first, with additional bells & whistles like sorting and pagination coming later - is that OK?

### CODE REVIEW

File: packages/svelte-materialify/src/components/DataTable/_variables.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+@import '../../styles/variables';
+
+$tbl-border-color: rgba(0, 0, 0, 0.12);
+$tbl-border-radius: $border-radius-root !default;
+$tbl-border-width: 1px;
+$tbl-background-hover: rgba(0, 0, 0, 0.04);
+$tbl-head-font-weight: map-get($font-weights, 'medium');
```
Comment (marekdedic) [inline packages/svelte-materialify/src/components/DataTable/_variables.scss:7]: Anyone reviewing this: Could you please check this? Especially the colors and how they interact with the dark theme...
Thanks

**Comment by marekdedic** [comment]
This should be basically ready. I've gone very "light touch", added just the very basic functionality and not much space for customization.

The general structure is in place, however, and should be pretty easy to extend.

**Comment by Florian-Schoenherr** [comment]
This is pretty neat! I hope someone else with experience on data table usage can review.

I only noticed that lines on dark mode are not really noticable:
dark mode lines

**Comment by marekdedic** [comment]
Dabbled in it some more, should look fine now ;)

**Review by Florian-Schoenherr — APPROVED** [review]

**Comment by Florian-Schoenherr** [comment]
@marekdedic I could merge this, too, again pointing to the rewrite of the lib. It's high demand and does what it should, the only thing I wonder about (haven't used DataTables really):
arch
looks like a lot to write..
But I just looked at the code, it seems useful for later building out the other usecases and for simple usecases we can provide wrappers.

**Comment by marekdedic** [comment]
Yes, I thought about that too - whether to do custom components, or just use something like `<tr>` and `<td>` - in the end, I was thinking along the same lines - using custom elements keeps the options open to do more exciting stuff in the future. It's a lot to write, but the rest of the library seems to use this style (as opposed to it being just `<Row>` and `<Cell>`) and you can always just

```js
import {DataTable, DataTableRow as Row, DataTableCell as Cell} from 'svelte-materialify';
```

**Comment by Florian-Schoenherr** [comment]
@marekdedic I forgot about that.
`class` props are missing on the types, after these are added I will merge.

**Comment by marekdedic** [comment]
Good catch, added them.

**Comment by marekdedic** [comment]
Hi,
could we please get this in a release? I'm not gonna do anything more with DataTable since svelte-materialify is getting deprecated :/

**Comment by Florian-Schoenherr** [comment]
I can't release, maybe @TheComputerM will do it

**Pull request merged**